### PR TITLE
AppVeyor: add GitHub tags on release, removes commented step

### PR DESCRIPTION
**Pull request by mauroservienti**
Connected to https://github.com/RadicalFx/housekeeping/issues/47

### CODE REVIEW

**Comment by mauroservienti** [comment]
@micdenny I'd love to have your input here, the [AppVeyor guide](https://www.appveyor.com/docs/deployment/github) utilizes this syntax to name a release on GitHub:

```
deploy:
  release: myproduct-v$(appveyor_build_version)
```

We are only interested in using the version number, I tried

```
deploy:
  release: %GitVersion_NuGetVersion%
```

And it fails stating that the `%` is not a valid char, so I changed it to

```
deploy:
  release: '%GitVersion_NuGetVersion%'
```

And the build works, obviously I have no easy way to test it.

**Comment by mauroservienti** [comment]
it's not that easy...I'm testing this on a side project. Marking as WIP, do not merge.

**Comment by mauroservienti** [comment]
OK, got it. We need to add 2 things:

* `skip_tags: true` so that a build is not triggered when adding a tag, otherwise as we create a release on GitHub a new build is triggered.
* use `$(env_variable)` to expand an environment variable

Removing WIP, this is ready to be merged. @micdenny please review.

File: appveyor.yml [hunk 1]
```diff
@@ -28,9 +29,6 @@
     - md "src\build\nuget"
     - nuget pack "src\nuget\Radical.Windows.Presentation.CastleWindsor.nuspec" -Version "%GitVersion_NuGetVersion%" -Properties "configuration=%CONFIGURATION%" -OutputDirectory "src\build\nuget"
 
-#  pull_requests:
-#    do_not_increment_build_number: true
```
Comment (micdenny) [inline appveyor.yml]: ok to remove this as it is useless now because `GitVersion` is renaming the AppVeyor version anyway

File: appveyor.yml [hunk 2]
```diff
@@ -86,3 +82,11 @@
     name: NuGet public feed
     on:
       branch: master
+  - provider: GitHub    
+    release: $(GitVersion_NuGetVersion)
+    description: 'Release v$(GitVersion_NuGetVersion)'
+    auth_token:
+      secure: 2ijF4OfPYfIUTunubL/Z04ogADL9sx/gKW1jMRNJnAa1OpHZpSvCKmP0gvNs8C7h
```
Comment (micdenny) [inline appveyor.yml]: and this of course is encrypted with a private key exchanged only with appveyor, right?

**Comment by micdenny** [comment]
after that couple of comments I left, to me you can merge and see if everything works as expected

File: appveyor.yml (hunk 1 above)
Comment (mauroservienti) [inline appveyor.yml]: :+1:

File: appveyor.yml (hunk 2 above)
Comment (mauroservienti) [inline appveyor.yml]: yes, correct. It is the `repo_public` access token for the `radical-bot` user encrypted by `AppVeyor`.

**Comment by mauroservienti** [comment]
merging.

**Pull request merged**